src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,21 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
         c.setTime(date);
+        String timeZone = getCustomIdOfTimeZone(sessionTimeZone);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
```

Comment:
Suppose the original time in UTC was `1st Jan 2016 1200 UTC NOON` , and the desired time zone is +1 GMT, your method would output something like `1st Jan 1300 +0100 NOON`, which would not be what is expected.